### PR TITLE
[IMP] base: ir_config_parameter, use default value if not found

### PR DESCRIPTION
**Pull request by JKE-be**
Follow docstring of get_param:
    The value of the parameter, or ``default`` if it does not exist.

Before this commit, if the parameter contains a falsy value, the default value
was returned.

Now, we only return the default value if it is None, the value returned by
_get_param in case of missing parameters.

Goal:

If you have a ir.config_parameter eg 'module.prefix' that you set to empty,
you should be able to do ICP.get_param('module.prefix') + '/ok' without add
manually add a default value (..., '').

### CODE REVIEW

**Comment by odony** [comment]
From what I understand, your point is to be able to store (and retrieve) empty strings in system parameters. 

The patch could make sense, but the rationale for doing it looks dangerous to me for a few reasons:

1) Your code will break hard if the parameter is missing. You can't exclude the possibility that it can happen, just like `env.ref(xml_id)` can yield no record, and we generally want to be defensive there. You could argue that system parameters are low-level stuff and one can break everything by deleting them (e.g. the few ones defined as default_parameters). That's true, but not all parameters are so sensitive. In your case we're talking of a parameter that is empty, so it should be considered optional, really.

```py
In [1]: env['ir.config_parameter'].get_param('my.prefix') + '/web/content'
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-10-b6a0e921337f> in <module>()
----> 1 env['ir.config_parameter'].get_param('my.prefix') + '/web/content'

TypeError: unsupported operand type(s) for +: 'bool' and 'str'

In [2]: env['ir.config_parameter'].get_param('my.prefix', '') + '/web/content'
Out[2]: '/web/content'
```

2) The `value` field of system parameters is *required*, so the web client won't allow you to save an empty string in the UI unless you remove that flag in the view. Which is weird because the value *is* indeed required. Historically we have always considered that empty Char/Text fields were supposed to be saved as NULL (because users can't make the difference between NULL and `''`). Similarly, in a search view, filtering on "Name is not set" will translate into a `[('name', '=', False)]`, not `[('name', '=', '')]`. We support storing empty strings via business code, but it is not something we want to encourage.

3) Most of our system parameters are *not* meant to be empty, and our code is generally not prepared to deal with that. If we allow empty values (e.g. with the hack mentioned in previous point), it means places where we *did* pass a default param and *did not* expect empty values may suddenly start to receive an empty string instead of the default. As a result we may have to adapt those existing callers to be more defensive, and that defeats the purpose of the `default` param a little bit:

```py
# previously this was safe:
jitsi_domain = ICP.get_param('website_jitsi.jitsi_server_domain', 'meet.jit.si')

# if we allow empty strings, we need to be more defensive, can't safely use the default anymore
jitsi_domain = ICP.get_param('website_jitsi.jitsi_server_domain') or 'meet.jit.si'
```

To summarize my thoughts: not officially supporting empty strings for system parameters made our business code simpler and safer in most cases, without any major drawbacks, because we don't generally need empty parameters (what's the point of an empty value - might as well be deleted). If the only reason for this change is to be able to save a few characters by not passing `default=''` to `get_param()` in a couple of places, it seems to bring more potential problems than solutions.
Why not make a helper function for your specific parameter instead? The full version of `env['ir.config_parameter'].sudo()._get_param('module.prefix')` is already quite verbose and ugly anyway ;-)

Unless you had more extensive cases in mind?

**Comment by JKE-be** [comment]
Thank for this detailed reply.

Purpose was to fix a report where user complains that without delete the web.base.url, but update it to '', the code of website will crash due to wrong concatenation of web.base.url to another path. (Delete the web.base.url is probably the worst thing to support in Odoo since it is used in 100+ place without default value. joker: low-level stuff arguments).

But at the end, what I probably miss, is that the value was required. So I'm agree that it doesn't make sense to support `''` if you update directly in db, or update view, it is at your own risk.

So, in worst case, I will just patch the code from the helper `get_base_url` that should be used everywhere instead of the ugly `env['ir.config_parameter'].sudo()._get_param('web.base.url')`. But it will just delay the crash, because I continue to think that odoo without a 'web.base.url' ICP will just not works (and is the expected result, we don't want to support it).

To conclude, I should just respect my own rule ! Never believe what user says, only what you see...
My bad. And thanks for your time.

**Comment by odony** [comment]
Thanks for the extra details! :pray:
